### DISCUSSION TITLE
Métriques - Considération des DEL

### DISCUSSION THREAD

**Discussion by r-carpentier** [beng-git/DARC2022#5] [Q&A] [answered]
Bonjour,

Les scripts de calcul des métriques fournis sur Github ne prennent pas en compte le fait que des tuples sont marqués comme DEL dans leur identifiant. Ainsi les métriques calculées restent de 1.0 (sauf tile) malgré la suppression d'un grand nombre de tuples.

De plus, l'utilité calculée dans ce cas semble différer de celle du serveur sds5000. Par exemple si je marque comme DEL les 30 millions de premiers tuples j'obtiens une utilité de 1 sur date, distance, hour, meet, poi et 0.1018 sur tile soit une moyenne de 0.85 alors que le serveur m'annonce une utilité de 0.5411.
(Voir dataset 53 nommé testAvec30MdeDEL de l'équipe petrus)

Ai-je omis quelque chose ?

**Comment by beng-git** [comment]
Merci on va regarder, le score devrait en effet plutôt être autour de 0.

**Comment by nancy-awad** [comment, marked as answer]
En fait il fallait expliquer le pré-traitement des données qui se passe avant le calcul des métriques.
1)	Tous les records avec le marquage ‘DEL’ sont supprimés du dataset anonymisé avant que celui-ci est passé en argument aux différentes métriques d’utilité. En d’autres termes si vous décidez de supprimer 70% des records en précisant ’DEL’ pour l’ID, toutes les métriques d’utilité définies reçoivent en paramètre les 30% des records du dataset anonymisé qui restent non supprimés.

2)	Les valeurs des colonnes dans ces records supprimés ne sont pas importantes. Il suffit de respecter le type de la colonne ou même laisser les valeurs initiales.

3)	Les différentes métriques ne reçoivent pas exactement le même nombre et type d’arguments :

- Les métriques {POI, tuile et meet } doivent recevoir en paramètres :
i.	Le dataset anonymisé (certainement sans les records ‘DEL’)
ii.	Le dataset original brut

- Les métriques {Date, Hour, Distance} doivent recevoir en paramètres :
i.	Le dataset anonymisé (certainement sans les records ‘DEL’)
ii.	Le dataset original excluant les records qui correspondent aux records ‘DEL’ dans le dataset anonymisé.
(Ce qui fait que vous devez avoir le même nombre de records pour les deux datasets)
iii.	Le nombre total de records original (pour pouvoir calculer la moyenne en tenant compte les records ‘DEL’ ( Ainsi leurs utilités devient égal a 0 automatiquement))

4) Il y avait un Bug sur le serveur qui est corrigé ainsi que **d'autres changements expliqués dans la nouvelle version du jupyter notebook**

**Reply by r-carpentier** [reply]
C'est compris, merci !
